extensions/amp-next-page/0.1/next-page-service.js

```diff
@@ -497,19 +516,57 @@ export class NextPageService {
 
   /**
    * Sets the specified document as active, updating the document title and URL.
-   * @param {!DocumentRef} documentRef Reference to the document to set as
-   *     active.
+   *
+   * @param {DocumentRef} ref Reference to the document to be activated
    * @private
    */
-  setActiveDocument_(documentRef) {
-    const {amp} = documentRef;
-    this.win_.document.title = amp.title || '';
-    this.activeDocumentRef_ = documentRef;
-    this.setActiveDocumentInHistory_(documentRef);
+  setActiveDocument_(ref) {
+    this.documentRefs_.forEach(docRef => {
+      const {amp} = docRef;
+      // Update the title and history
+      if (docRef === ref) {
+        this.win_.document.title = amp.title || '';
+        this.activeDocumentRef_ = docRef;
+        this.setActiveDocumentInHistory_(docRef);
+        // Show the active document
+        this.setDocumentVisibility_(docRef, VisibilityState.VISIBLE);
+      } else {
+        // Hide other documents
+        this.setDocumentVisibility_(docRef, VisibilityState.HIDDEN);
+      }
+    });
 
     // TODO(emarchiori): Consider updating position fixed elements.
   }
 
+  /**
+   * Manually overrides the document's visible state to the given state
+   *
+   * @param {DocumentRef} ref Reference to the document to change
+   * @param {!../../../src/visibility-state.VisibilityState} visibilityState
+   * @private
+   */
+  setDocumentVisibility_(ref, visibilityState) {
+    // Prevent updating visibility of the host document
+    if (ref === this.documentRefs_[0]) {
+      return;
+    }
+
+    const ampDoc = ref && ref.amp && ref.amp.ampdoc;
```

Comment:
`ref` should be non-nullable here, since you are passing an entry from an array of non-nullable `DocumentRefs`. Can change the type and remove the first part of this.